qemu/tests/timerdevice_boot.py

```diff
@@ -42,6 +44,48 @@ def verify_guest_clock_source(session, expected):
             raise error.TestFail(
                 "Guest didn't use '%s' clocksource" % expected)
 
+    def verify_base_date_system_time():
+        """
+        When setting rtc_base to be a date,verify the guest system should be this date
+        """
+        if params["os_type"] == "linux":
+            (host_time, guest_time) = utils_test.get_time(session,
+                                                          "%s, -u" % time_command,
+                                                          time_filter_re, time_format)
+        else:
+            (host_time, guest_time) = utils_test.get_time(session, time_command,
+                                                          time_filter_re, time_format)
+        guest_time = time.strftime('%Y-%m-%d %H', time.localtime(guest_time))
+        logging.info("guest_time:%s", guest_time)
+        if guest_time != '%s 00' % params["rtc_base"]:
+            session.close()
+            test.fail("Guest time is %s and not from %s 00:00"
+                      % (guest_time, params["rtc_base"]))
+
+    def verify_guest_system_time():
+        error_context.context("Check the system time on guest and host", logging.info)
+        (host_time, guest_time) = utils_test.get_time(session, time_command,
+                                                      time_filter_re, time_format)
+        drift = abs(float(host_time) - float(guest_time))
+        if drift > timerdevice_drift_threshold:
+            session.close()
+            test.fail("The guest's system time is different with"
+                      " host's. Host time: '%s', guest time:"
+                      " '%s'" % (host_time, guest_time))
+
+    def verify_guest_hw_time():
+        get_hw_time_cmd = params.get("get_hw_time_cmd")
+        if get_hw_time_cmd:
+            error_context.context("Check the hardware time on guest and host", logging.info)
+            host_time = utils.system_output(get_hw_time_cmd)
+            guest_time = session.cmd(get_hw_time_cmd)
+            drift = abs(float(host_time) - float(guest_time))
+            if drift > timerdevice_drift_threshold:
+                session.close()
+                test.fail("The guest's hardware time is different with"
+                          " host's. Host time: '%s', guest time:"
+                          " '%s'" % (host_time, guest_time))
```

Comment:
Line 82-87 can be combined with line 69-74 as a function like compare_time(host_time, guest_time, type="hardware/system")

Reply:
Combine verify_guest_hw_time() and verify_guest_system_time() to verify_guest_time(), add a function compare_time() to check the system/hardware time for host and guest.